mininet/net.py

```diff
@@ -729,9 +729,10 @@ def wifiDirect(self, sta, cls=None, **params):
             \ndevice_name=%s\
             \ndevice_type=1-0050F204-1\
             \np2p_no_group_iface=1' % (sta)
-        apcommand = cmd + ("\' > %s_wifiDirect.conf" % sta)
+        confname = "mn%d_%s_wifiDirect.conf" % (os.getpid(), sta)
+        apcommand = cmd + ("\' > %s" % confname)
         os.system(apcommand)
-        sta.cmd('wpa_supplicant -B -Dnl80211 -i%s-wlan0 -d -c%s_wifiDirect.conf' % (sta, sta))
+        sta.cmd('wpa_supplicant -B -Dnl80211 -c%s -i%s-wlan0 -d' % (confname, sta))
```

Comment:
There are more parts in the code where always interface 0 is used. But this is not my change.